sympy/sets/ordinals.py

```diff
@@ -0,0 +1,131 @@
+from sympy.core import Basic
+
+def reverse_bisect_right(a, x, lo=0, hi=None):
```

Comment:
The use of binary search looks like overkill to me, at least in the beginning phase. In any case, I think that it is not necessary to include the name of the implementation method in the name of the function.